### PR TITLE
feat(project-operator): [KDL6-186] Improve project operator chart structure

### PR DESCRIPTION
**Pull request by ialejandro**
## Motivation and Context

This change aims to improve the maintainability and organization of the KDL Project Helm chart. The current structure has components mixed together, making it difficult to maintain and understand the relationships between different parts of the system. By reorganizing the templates into dedicated directories and establishing a clearer separation of concerns, we make the chart more maintainable and easier to extend in the future.

The reorganization follows Helm best practices by:

1. Separating components into their own directories
2. Centralizing helper functions
3. Removing duplicate templates
4. Improving the overall structure for better clarity

## PR Checklist

- [X] I have created tests for my code changes, and the tests are passing.
- [X] I have executed the pre-commit hooks locally.
- [X] I have updated the documentation accordingly.
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added
- [X] My change requires a change to the documentation (already included in this PR)

## PR Type

- [X] Refactoring (no functional changes, no api changes)
- [X] Documentation content changes

## What is the current behavior?

Currently, the Helm chart structure has all templates mixed in a single directory, making it difficult to:
* Understand component boundaries
* Maintain individual components
* Make changes to one component without affecting others
* Navigate and find specific templates

## What is the new behavior?

The new structure:

1. Organizes templates by component:
   * `templates/filebrowser/` for Filebrowser-related templates
   * `templates/mlflow/` for MLflow-related templates

2. Improves helper template organization:
   * Centralizes common helpers in `_helpers.tpl`
   * Removes duplicate template definitions

3. Separates configuration:
   * Individual component configurations in their respective directories
   * Shared configurations in root level
   * CI-specific values in dedicated files

4. Updates documentation:
   * Clearer README.md
   * Updated NOTES.txt with new structure
   * Better inline documentation

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Other information

### CODE REVIEW

**Review by igzjaviergil — APPROVED** [review]

File: project-operator/helm-charts/kdl-project/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+sources:
+  - https://github.com/konstellation-io/kdl-server
+home: https://www.konstellation.io
+maintainers:
+  - name: ialejandro
+    email: ivan.alejandro@intelygenz.com
+  - name: alpiquero
+    email: angelluis.piquero@intelygenz.com
+  - name: danielchg
+    email: daniel.chavero@intelygenz.com
```
[the remark this answers is not in the document]
Comment (ialejandro) [reply project-operator/helm-charts/kdl-project/Chart.yaml:16]: Daniel nope,but I like remember those who have fallen in combat.

File: app/api/infrastructure/k8s/client.go [hunk 2]
```diff
@@ -20,13 +20,13 @@ const (
...
 	userToolsAPIVersion = userToolsGroup + "/" + userToolsVersion
 
-	kdlprojectGroup      = "project.konstellation.io"
+	kdlprojectGroup      = "kdl.konstellation.io"
 	kdlprojectResource   = "kdlprojects"
 	kdlprojectVersion    = "v1"
 	kdlprojectAPIVersion = kdlprojectGroup + "/" + kdlprojectVersion
 )
 
-var _ ClientInterface = &Client{}
```
Comment (alexrodfe) [inline app/api/infrastructure/k8s/client.go:29]: any reason for this change?

this var is written in this specific manner to demonstrate Client structure implements ClientInterface

Comment (igzjaviergil) [reply app/api/infrastructure/k8s/client.go:29]: I updated to use more generic way to assert that Client structure implements ClientInterface with `var _ ClientInterface = (*Client)(nil)`

File: app/api/infrastructure/k8s/kdlproject.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ func (k *Client) CreateKDLProjectCR(ctx context.Context, projectID string) error
 	}
 
 	mlflowEnv["ARTIFACTS_BUCKET"] = projectID
-	// FUTURE: update AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY with minIO values for the project
+	// FUTURE: update MINIO_ACCESS_KEY and MINIO_SECRET_KEY with minIO values for the project
```
Comment (ialejandro) [inline app/api/infrastructure/k8s/kdlproject.go:53]: I don't understand this change. MLflow uses `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`. Ref: <https://mlflow.org/docs/latest/tracking/artifacts-stores.html>, I think we should follow standard variables.

Comment (igzjaviergil) [reply app/api/infrastructure/k8s/kdlproject.go]: my bad, I got confused with all the different var names

**Review by alexrodfe — APPROVED** [review]
Great work!

**Pull request merged**